src/common/commonutils/UrlUtils.c

```diff
@@ -32,7 +34,7 @@ char* UrlEncode(char* target)
             }
             else
             {
-                sprintf(&encodedTarget[j], "%%%02X", target[i]);
+                snprintf(&encodedTarget[j], encodedLength - j, "%%%02X", target[i]);
                 j += strlen(&encodedTarget[j]);

```

Comment:
Will it be different than 3 if we allocated proper size for the buffer?

Reply:
As explained, not changing any of this now, risk of regression is too high.